src/client-builder.ts

```diff
@@ -176,7 +176,8 @@ export class ClientBuilder<Resources extends ResourceTypeConstraint> {
           const chainResponsePhase =
             (previousValue: ResponseGetter, currentValue: MiddlewareDescriptor) => () => {
               // Deliberately putting this on two separate lines - to get typescript to not return "any"
-              const nextValue = currentValue.response(previousValue, renew)
+              const getFinalRequest = () => finalRequest
+              const nextValue = currentValue.response(previousValue, renew, getFinalRequest)
```

Comment:
Why pass a function instead of the real value directly (`finalRequest`)?

Also, should we protect this of being mutated between `response`-calls in the chain? 🤔

Reply:
Definately looking over this in mappersmith v3. I imagine we return `{ next, request, renew }` for example